File: internal/content/testdata/bench/javascript/sort-array-of-three-values.js
function sort_three_values(arr, low_val = 0, mid_val = 1, high_val = 2) {
    let lo = 0, mid = 0, hi = arr.length - 1;
    while (mid <= hi) {
        if (arr[mid] === low_val) {
            [arr[lo], arr[mid]] = [arr[mid], arr[lo]];
            lo++;
            mid++;
        } else if (arr[mid] === mid_val) {
            mid++;
        } else {
            [arr[mid], arr[hi]] = [arr[hi], arr[mid]];
            hi--;
        }
    }
    return arr;
}

File: internal/content/testdata/bench/java/sort-array-of-three-values.java
class Solution {
    public long[] sort_three_values(long[] arr, long low_val, long mid_val, long high_val) {
        int lo = 0, mid = 0, hi = arr.length - 1;
        while (mid <= hi) {
            if (arr[mid] == low_val) {
                long tmp = arr[lo]; arr[lo] = arr[mid]; arr[mid] = tmp;
                lo++;
                mid++;
            } else if (arr[mid] == mid_val) {
                mid++;
            } else {
                long tmp = arr[mid]; arr[mid] = arr[hi]; arr[hi] = tmp;
                hi--;
            }
        }
        return arr;
    }
}
